pkg/landscaper/blueprints/store.go

```diff
@@ -0,0 +1,303 @@
+// SPDX-FileCopyrightText: 2021 SAP SE or an SAP affiliate company and Gardener contributors.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package blueprints
+
+import (
+	"context"
+	"encoding/base32"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"sync"
+	"time"
+
+	"github.com/gardener/component-cli/ociclient/cache"
+	cdv2 "github.com/gardener/component-spec/bindings-go/apis/v2"
+	"github.com/go-logr/logr"
+	"github.com/mandelsoft/vfs/pkg/memoryfs"
+	"github.com/mandelsoft/vfs/pkg/projectionfs"
+	"github.com/mandelsoft/vfs/pkg/readonlyfs"
+	"github.com/mandelsoft/vfs/pkg/vfs"
+	"k8s.io/apimachinery/pkg/api/resource"
+
+	"github.com/gardener/landscaper/apis/config/v1alpha1"
+
+	"github.com/gardener/landscaper/pkg/utils/tar"
+
+	"github.com/gardener/landscaper/apis/config"
+
+	lsv1alpha1 "github.com/gardener/landscaper/apis/core/v1alpha1"
+	"github.com/gardener/landscaper/pkg/api"
+	"github.com/gardener/landscaper/pkg/utils"
+)
+
+var storeSingleton *Store
+var mux sync.Mutex
+
+// GetStore returns the currently active store.
+func GetStore() *Store {
+	mux.Lock()
+	defer mux.Unlock()
+	return storeSingleton
+}
```

Comment:
What do you want to achieve with the mutex in this method? You will still conflict with SetStore because you get a pointer to a Store which might be closed just after you fetched it.  But perhaps I misunderstood this part.

Reply:
Yes you're right I could get a pointer to a closed store.
Maybe I also need to check that and return errors when the store is already closed.

As for the mutex: I first wanted to have a real singleton like 
```
func GetStore() *Store {
  if store == nil {
    store = createNewStore()
  }
  return store
}
```
In that case a mutex was necessary to no create multiple stores at once.
But I decided against it as the store needs more error handling like temp dirs defaulting etc..
Thats why a default store is not initialized in the init function.

But after that you're right that the new mutex does not really give any more safety.
I will remove it and will return proper errors for closed stores